src/Features/Core/Completion/Snippets/ISnippetCompletionListProvider.cs

```diff
@@ -0,0 +1,8 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Completion.Snippets
+{
+    internal interface ISnippetCompletionListProvider
```

Comment:
Should this inherit from some interface just so the type system can help us out here? It's unclear to me what (if any) types this can be applied to or what such an application means. XML doc comments might just help.